code/__defines/observation.dm

```diff
@@ -0,0 +1,7 @@
+/// Used to trigger observations and call procs registered for that observation
+/// The datum hosting the observation is automaticaly added as the first argument
+/// Returns a bitfield gathered from all registered procs
+/// Arguments given here are packaged in a list and given to _RaiseEvenet
+#define RAISE_EVENT(listener, obstype, arguments...) ( !listener.observations || !listener.observations[obstype.name] ? NONE : listener._RaiseEvent(obstype, list(listener, ##arguments)) )
```

Comment:
Listener is being used wrongly here, this is the source not the listener

Reply:
Don't take shortcuts with variable names, use observation_datum since what's being passed here is a datum ref, not a typepath